api/src/com/cloud/network/vpc/OSPFZoneConfig.java

```diff
@@ -0,0 +1,329 @@
+            }
+            this.superCIDRList = v_cidr;
+        }
+        if (enabled != null) {
+            this.enabled = enabled;
+        }
+    }
+
+    public void setValues(Map<String, String> details) throws BadCIDRException {
+        Protocol protocol = Protocol.valueOf(details.get(Params.PROTOCOL.name()));
+        String ospfArea = details.get(Params.AREA.name());
+        Short helloInterval = Short.valueOf(details.get(Params.HELLO_INTERVAL.name()));
+        Short deadInterval = Short.valueOf(details.get(Params.DEAD_INTERVAL.name()));
+        Short retransmitInterval = Short.valueOf(details.get(Params.RETRANSMIT_INTERVAL.name()));
+        Short transitDelay = Short.valueOf(details.get(Params.TRANSIT_DELAY.name()));
+        Authentication authentication = Authentication.valueOf(details.get(Params.AUTHENTICATION.name()));
+        String quaggaPassword = details.get(Params.PASSWORD.name());
+        String superCIDRList = details.get(Params.SUPER_CIDR.name());
+        Boolean enabled = Boolean.valueOf(details.get(Params.ENABLED.name()));
+
+        if (protocol != null) {
+            if (Protocol.OSPF == protocol) {
+                this.protocol = protocol;
+            } else {
+                throw new InvalidParameterValueException("The quagga protocal can have values of OSPF  " + protocol);
+            }
+        }
+        if (ospfArea != null) {
+            this.ospfArea = ospfArea;
+        }
+        if (helloInterval != null) {
+            this.helloInterval = helloInterval;
+        }
+        if (deadInterval != null) {
+            this.deadInterval = deadInterval;
+        }
+        if (retransmitInterval != null) {
+            this.retransmitInterval = retransmitInterval;
+        }
+        if (transitDelay != null) {
+            this.transitDelay = transitDelay;
+        }
+        if (authentication != null) {
+            if (Authentication.MD5 == authentication || Authentication.PLAIN_TEXT == authentication) {
+                this.authentication = authentication;
+            } else {
+                throw new InvalidParameterValueException("The quagga authentication can have values of MD5 or PLAIN_TEXT and not " + authentication);
+            }
+        }
+        if (quaggaPassword != null) {
+            this.password = quaggaPassword;
+        }
+        if (superCIDRList != null) {
+            String[] cidr_list = superCIDRList.split(",");
+            for (String cidr : cidr_list) {
+                if (!NetUtils.isValidCIDR(cidr)) {
+                    throw new InvalidParameterValueException("The super CIDR is not a valid cidr " + cidr);
+                }
+            }
+            CIDR[] v_cidr = NetUtils.convertToCIDR(cidr_list);
+            if (!NetUtils.cidrListConsistency(v_cidr)) {
+                throw new InvalidParameterValueException("The cidr list is not consistent " + Arrays.toString(cidr_list));
+            }
+            this.superCIDRList = v_cidr;
+        }
```

Comment:
Lines 261-273 appear to be a repeat of 199-211.  Please to a common method or consider integrating into the ``CIDR`` collection class discussed earlier.